build-packages.js

```diff
@@ -2,48 +2,73 @@ const fs = require("fs");
 const path = require("path");
 const del = require("del");
 const util = require("util");
+const { kebabCase, map } = require("lodash");
+const { peerDependencies } = require("./package.json");
 
 // Convert NodeJS methods to promises in order to use them with `await` statement
 const exec = util.promisify(require("child_process").exec);
 const writeFile = util.promisify(fs.writeFile);
+const makeDir = util.promisify(fs.mkdir);
 
 // Read available package sources from the dir
 const entryDirPath = path.join(__dirname, "src/packages");
 
 fs.readdir(entryDirPath, async (e, files) => {
-  console.log(`⚙️ Building ${files.length} packages...`);
-
-  for (let file of files) {
+  // Bundles a package asynchronously
+  const bundlePackage = async (file) => {
     const { name } = path.parse(file);
     const isMainPackage = name === "hex";
-    const filePath = path.join(entryDirPath, file);
     const outputDirPath = path.join(__dirname, isMainPackage ? "dist" : name);
     const manifestPath = path.join(outputDirPath, `package.json`);
     const bundlerPath = path.join(__dirname, "node_modules/.bin/microbundle");
 
+    // Format a package name according to NPM's naming guide
+    // https://docs.npmjs.com/files/package.json#name
+    const packageName = `react-colorful-${kebabCase(name)}`;
+
     // Delete the previous package version if exists
     await del(outputDirPath);
 
-    // Run microbundle
-    const { stdout } = await exec(
-      `${bundlerPath} --entry ${filePath} --output ${outputDirPath}/index.js  --name react-colorful-${name} --css-modules true --jsx React.createElement`
-    );
-    console.log(stdout);
-
     if (!isMainPackage) {
       // Create `package.json`
       var manifestCode = JSON.stringify({
-        name: `react-colorful-${name}`,
+        name: packageName,
         private: true,
         main: "index.js",
         module: "index.module.js",
         esmodule: "index.esmodule.js",
-        types: "../types/index.d.ts"
+        "umd:main": "dist/index.umd.js",
```

Comment:
Oh, and this doesn't look correct. Sorry, didn't catch it.

Reply:
Not sure that I got what you mean)

Reply:
The `dist`. If you try to use the `umd:main` you'll be pointed at a file that doesn't exist. Should be `index.umd.js`.

Reply:
Microbundle still generates the right file, it's just the path that's incorrect.

Reply:
Oh, I see. Nice catch, thanks!